beluga-aws-sdk/src/jobs/mod.rs

```diff
@@ -0,0 +1,238 @@
+use std::collections::HashMap;
+
+use beluga_mqtt::{MqttClient, QoS};
+use chrono::prelude::Utc;
+use futures::prelude::future::FutureExt;
+
+use self::objects::{
+    GetPendingJobsExecutionReq, GetPendingJobsExecutionResp, JobExecution, JobExecutionSummary,
+    JobStatus, StartNextPendingJobExecutionReq, StartNextPendingJobExecutionResp,
+};
+use crate::error::Error;
+use crate::Result;
+
+mod datetime;
+mod objects;
+
+#[derive(Debug)]
+pub struct JobsClient {
+    mqtt: MqttClient,
+}
+
+impl JobsClient {
+    pub const fn new(mqtt: MqttClient) -> Self {
+        Self { mqtt }
+    }
+
+    pub async fn get(&self) -> Result<(Vec<Job>, Vec<Job>)> {
+        self.mqtt
+            .publish(
+                format!("$aws/things/{}/jobs/get", self.mqtt.thing_name()).as_str(),
+                QoS::AtLeastOnce,
+                false,
+                bytes::Bytes::copy_from_slice(&serde_json::to_vec(&GetPendingJobsExecutionReq {
+                    token: format!("get-jobs-{}", self.mqtt.thing_name()),
+                })?),
+            )
+            .await?;
+
+        let accepted = format!("$aws/things/{}/jobs/get/accepted", self.mqtt.thing_name());
+        let rejected = format!("$aws/things/{}/jobs/get/rejected", self.mqtt.thing_name());
+
+        let mut subscriber = self
+            .mqtt
+            .subscribe_many([accepted.as_str(), rejected.as_str()], QoS::AtLeastOnce)
+            .await?;
```

Comment:
we should subscribe before publishing to avoid a race condition.

Does the aws sdk also only temporarily subscribe to the accepted and rejected topics? I would've assumed the subscribes to these topics once at the start.